src/libs/model/functions.ts

```diff
@@ -0,0 +1,187 @@
+import {
+  IElement,
+  IPackage,
+  IClassifier,
+  IGeneralization,
+  IContainer,
+} from '@types';
+import { OntoUMLType } from '@constants/.';
+import memoizee from 'memoizee';
+
+export default {
+  IElement_functions: {
+    getRootPackage: getRootPackage,
+    // getRootPackage: memoizee(getRootPackage),
+    hasIContainerType,
+    hasIDecoratableType,
+    hasIClassifierType,
+  },
+  IContainer_functions: {
+    getAllContents: getAllContents,
+    // getAllContents: memoizee(getAllContents),
+    getAllContentsByType: getAllContentsByType,
+    // getAllContentsByType: memoizee(getAllContentsByType),
+    getContentById: getContentById,
+    // getContentById: memoizee(getContentById),
+  },
+  IClassifier_functions: {
+    getParents: getParents,
+    // getParents: memoizee(getParents),
+    getChildren: getChildren,
+    // getChildren: memoizee(getChildren),
+    getAncestors: getAncestors,
+    // getAncestors: memoizee(getAncestors),
+    getDescendents: getDescendents,
+    // getDescendents: memoizee(getDescendents),
+  },
+};
+
+function getRootPackage(): IPackage {
+  const self = this as IElement;
+
+  if (self.container) {
+    let root: IPackage;
```

Comment:
You can do in one line like this:

`const root: IPackage = (self.container as IContainer).getRootPackage();`